### PR TITLE
BUG-105281 Provide extra info for while Ambari failed

### PR DESCRIPTION
**Pull request by daszabo**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariAdapter.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                .filter(component -> !CLIENT_CATEGORY_NAME.equalsIgnoreCase(component.category))
+                .map(ClusterComponent::getState)
+                .collect(Collectors.toList()));
+
+        Optional<ClusterStatus> optPartialStatus = clusterStatuses.stream().filter(partialStatuses::contains).findFirst();
+        if (optPartialStatus.isPresent()) {
+            ClusterStatus clusterStatus = optPartialStatus.get();
+            String componentsInStatus = reduceToHostComponentString(clusterComponents, clusterStatus);
+            return new ClusterStatusResult(clusterStatus, componentsInStatus);
+        } else if (clusterStatuses.stream().anyMatch(fullStatuses::contains) && clusterStatuses.size() == 1) {
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariAdapter.java]: Size check is cheaper than anymatch

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariAdapter.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            String componentsInInstalled = reduceToHostComponentString(clusterComponents, ClusterStatus.INSTALLED);
+            String componentsInUnsupported = reduceToHostComponentString(clusterComponents, ClusterStatus.AMBIGUOUS);
+            return new ClusterStatusResult(ClusterStatus.AMBIGUOUS, String.join("; ", componentsInInstalled, componentsInUnsupported));
+        }
+    }
+
+    private String reduceToHostComponentString(List<ClusterComponent> clusterComponents, ClusterStatus clusterStatus) {
+        return clusterComponents.stream()
+                        .filter(component -> clusterStatus.equals(component.state))
+                        .filter(component -> !CLIENT_CATEGORY_NAME.equalsIgnoreCase(component.category))
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariAdapter.java]: One filter with && also works

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariAdapter.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        private final ClusterStatus state;
+
+        @JsonCreator
+        ClusterComponent(@JsonProperty("host") String host,
+                @JsonProperty("component_name") String componentName,
+                @JsonProperty("state") String state,
+                @JsonProperty("category") String category) {
+            this.host = host;
+            this.componentName = componentName;
+            if (ClusterStatus.supported(state)) {
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariAdapter.java]: Elvis operator

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariAdapter.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sequenceiq.ambari.client.AmbariClient;
+import com.sequenceiq.cloudbreak.service.cluster.flow.status.ClusterStatus;
+
+@Component
+public class AmbariAdapter {
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariAdapter.java]: Missing test

Comment (daszabo) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariAdapter.java]: 👍

**Pull request merged**